### ISSUE TITLE
In Dev Preview, Storage upload progress `transferredBytes` larger than `totalBytes`

### ISSUE THREAD

**Issue by vgribok** [aws-amplify/amplify-flutter#2591]
### Description

Using Dev Preview amplify_storage_s3 version "1.0.0-next.3", observing that `fractionCompleted` goes to 1,500% (15.xxx) due to `transferredBytes` going way above the size of the file.

My code:
```dart
final StorageUploadFileResult<StorageItem> result = await uploadFile(
        localFile: localFile.toAwsFile(contentType: "image/jpeg"),
        options: const S3UploadFileOptions(accessLevel: StorageAccessLevel.private),
        key: s3Key,
        onProgress: (progress) {
          final bytesSent = progress.transferredBytes;
          final fileLength = progress.totalBytes;
          'I: Uploaded bytes: ${bytesSent.asThouCommas} of ${fileLength.asThouCommas}'
              ' (${(progress.fractionCompleted * 100).round()}%), of the "${localFile.baseNameWithExtension}"'.debugPrint();
        },
      ).result;
```
outputs:
> flutter: I: Uploaded bytes: 65,536 of 1,483,806 (4%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 196,608 of 1,483,806 (13%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 393,216 of 1,483,806 (27%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 655,360 of 1,483,806 (44%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 983,040 of 1,483,806 (66%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 1,376,256 of 1,483,806 (93%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 1,835,008 of 1,483,806 (124%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 2,359,296 of 1,483,806 (159%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 2,949,120 of 1,483,806 (199%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 3,604,480 of 1,483,806 (243%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 4,325,376 of 1,483,806 (292%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 5,111,808 of 1,483,806 (345%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 5,963,776 of 1,483,806 (402%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 6,881,280 of 1,483,806 (464%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 7,864,320 of 1,483,806 (530%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 8,912,896 of 1,483,806 (601%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 10,027,008 of 1,483,806 (676%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 11,206,656 of 1,483,806 (755%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 12,451,840 of 1,483,806 (839%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 13,762,560 of 1,483,806 (928%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 15,138,816 of 1,483,806 (1020%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 16,580,608 of 1,483,806 (1117%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 18,064,414 of 1,483,806 (1217%), of the "2023-01-21T21:07:35.822592.jpg"
flutter: I: Uploaded bytes: 18,064,414 of 1,483,806 (1217%), of the "2023-01-21T21:07:35.822592.jpg"

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [X] Storage

### Steps to Reproduce

1. Put a breakpoint in the `onProgress` of the Storage `upload()` method

### Platforms

- [X] iOS
- [ ] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Environment

```bash
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.10, on macOS 12.6.2 21G320 darwin-arm, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.1)
[✓] VS Code (version 1.74.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!
```

### Dependencies

```bash
Dart SDK 2.18.6eps --no-dev --style=compact                                                                                                                                                                                          ─╯
Flutter SDK 3.3.10
redacted

dependencies:
- amplify_api 1.0.0-next.3 [amplify_api_android amplify_api_ios amplify_core amplify_flutter async aws_common collection connectivity_plus flutter json_annotation meta plugin_platform_interface stream_transform web_socket_channel]
- amplify_auth_cognito 1.0.0-next.3 [amplify_auth_cognito_android amplify_auth_cognito_dart amplify_auth_cognito_ios amplify_core amplify_flutter amplify_secure_storage async flutter flutter_web_plugins meta path plugin_platform_interface]
- amplify_authenticator 1.0.0-next.2 [amplify_auth_cognito amplify_core amplify_flutter async aws_common collection flutter flutter_localizations intl meta smithy stream_transform]
- amplify_datastore 1.0.0-next.3 [flutter amplify_datastore_plugin_interface amplify_core plugin_platform_interface meta collection async]
- amplify_flutter 1.0.0-next.3 [amplify_core amplify_datastore_plugin_interface amplify_flutter_android amplify_flutter_ios amplify_secure_storage aws_common collection flutter meta plugin_platform_interface]
- amplify_storage_s3 1.0.0-next.3 [amplify_core amplify_db_common amplify_storage_s3_dart aws_common flutter meta path_provider]
- async 2.9.0 [collection meta]
- connectivity_plus 3.0.2 [flutter flutter_web_plugins connectivity_plus_platform_interface js meta nm]
- cupertino_icons 1.0.5
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_form_builder 7.7.0 [flutter intl collection]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_native_splash 2.2.16 [args flutter flutter_web_plugins js html image meta path universal_io xml yaml]
- flutter_nav2_oop 1.0.0+1 [flutter cupertino_icons flutter_riverpod google_fonts riverpod_restorable]
- flutter_riverpod 2.1.3 [collection flutter meta riverpod state_notifier]
- font_awesome_flutter 10.3.0 [flutter]
- form_builder_validators 8.4.0 [flutter flutter_localizations intl]
- geolocator 9.0.2 [flutter geolocator_platform_interface geolocator_android geolocator_apple geolocator_web geolocator_windows]
- image_picker 0.8.6 [flutter image_picker_android image_picker_for_web image_picker_ios image_picker_platform_interface]
- intl 0.17.0 [clock path]
- native_exif 0.4.0 [flutter intl]
- path 1.8.2
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- riverpod_restorable 2.0.0 [flutter riverpod flutter_riverpod]

transitive dependencies:
- amplify_api_android 1.0.0-next.3 [flutter]
- amplify_api_ios 1.0.0-next.3 [amplify_core flutter]
- amplify_auth_cognito_android 1.0.0-next.3 [flutter]
- amplify_auth_cognito_dart 0.4.0 [amplify_core amplify_secure_storage_dart async aws_common aws_signature_v4 built_collection built_value collection convert crypto fixnum http intl js json_annotation meta oauth2 path smithy smithy_aws stream_transform uuid worker_bee]
- amplify_auth_cognito_ios 1.0.0-next.3 [amplify_core flutter]
- amplify_core 1.0.0-next.3 [async aws_common aws_signature_v4 collection intl json_annotation logging meta retry uuid]
- amplify_datastore_plugin_interface 1.0.0-next.3 [amplify_core collection flutter meta]
- amplify_db_common 0.1.2+1 [amplify_db_common_dart drift flutter path path_provider]
- amplify_db_common_dart 0.2.0 [amplify_core async aws_common drift meta path sqlite3]
- amplify_flutter_android 1.0.0-next.3 [flutter]
- amplify_flutter_ios 1.0.0-next.3 [amplify_core flutter]
- amplify_secure_storage 0.1.4+1 [amplify_secure_storage_dart async file flutter meta path path_provider]
- amplify_secure_storage_dart 0.1.4+1 [async aws_common built_collection built_value ffi file js meta path win32 worker_bee]
- amplify_storage_s3_dart 0.1.4+1 [amplify_core amplify_db_common_dart async aws_common aws_signature_v4 built_collection built_value drift fixnum meta path smithy smithy_aws]
- archive 3.3.0 [crypto path]
- args 2.3.2
- aws_common 0.3.5+1 [async built_collection built_value collection http2 js json_annotation logging meta mime os_detect path stream_transform uuid]
- aws_signature_v4 0.3.1+1 [async aws_common collection convert crypto json_annotation meta path]
- built_collection 5.1.1
- built_value 8.4.3 [built_collection collection fixnum meta]
- characters 1.2.1
- clock 1.1.1
- collection 1.16.0
- connectivity_plus_platform_interface 1.2.3 [flutter meta plugin_platform_interface]
- convert 3.1.1 [typed_data]
- crclib 3.0.0 [meta tuple]
- cross_file 0.3.3+2 [js meta]
- crypto 3.0.2 [typed_data]
- csslib 0.17.2 [source_span]
- dbus 0.7.8 [args ffi meta xml]
- drift 2.3.0 [async convert collection js meta stream_channel sqlite3]
- ffi 2.0.1
- file 6.1.2 [meta path]
- fixnum 1.0.1
- flutter_plugin_android_lifecycle 2.0.7 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- geolocator_android 4.1.4 [flutter geolocator_platform_interface]
- geolocator_apple 2.2.3 [flutter geolocator_platform_interface]
- geolocator_platform_interface 4.0.7 [flutter plugin_platform_interface vector_math meta]
- geolocator_web 2.1.6 [flutter flutter_web_plugins geolocator_platform_interface]
- geolocator_windows 0.1.1 [flutter geolocator_platform_interface]
- google_fonts 3.0.1 [flutter http path_provider crypto]
- html 0.15.1 [csslib source_span]
- http 0.13.5 [async http_parser meta path]
- http2 2.0.1
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- image 3.3.0 [archive meta xml]
- image_picker_android 0.8.5+4 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface]
- image_picker_for_web 2.1.10 [flutter flutter_web_plugins image_picker_platform_interface]
- image_picker_ios 0.8.6+5 [flutter image_picker_platform_interface]
- image_picker_platform_interface 2.6.2 [cross_file flutter http plugin_platform_interface]
- js 0.6.4
- json_annotation 4.7.0 [meta]
- logging 1.1.0
- material_color_utilities 0.1.5
- meta 1.8.0
- mime 1.0.4
- nm 0.5.0 [dbus]
- oauth2 2.0.1 [collection crypto http http_parser]
- os_detect 2.0.1
- path_provider_android 2.0.22 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.11 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.7 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.5 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.3 [ffi flutter path path_provider_platform_interface win32]
- petitparser 5.1.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.3 [meta]
- process 4.2.4 [file path platform]
- retry 3.1.0
- riverpod 2.1.3 [collection meta stack_trace state_notifier]
- shelf 1.4.0 [async collection http_parser path stack_trace stream_channel]
- sky_engine 0.0.99
- smithy 0.4.0 [async aws_common built_collection built_value collection convert crypto fixnum http_parser intl json_annotation meta path retry shelf typed_data xml]
- smithy_aws 0.4.0 [aws_common aws_signature_v4 built_collection built_value collection convert crclib crypto intl json_annotation meta path smithy xml]
- source_span 1.9.0 [collection path term_glyph]
- sqlite3 1.9.1 [collection ffi js meta path]
- stack_trace 1.10.0 [path]
- state_notifier 0.7.2+1 [meta]
- stream_channel 2.1.0 [async]
- stream_transform 2.1.0
- string_scanner 1.1.1 [source_span]
- term_glyph 1.2.1
- tuple 2.0.1
- typed_data 1.3.1 [collection]
- universal_io 2.0.4 [collection crypto meta typed_data]
- uuid 3.0.7 [crypto]
- vector_math 2.1.2
- web_socket_channel 2.3.0 [async crypto stream_channel]
- win32 3.1.3 [ffi]
- worker_bee 0.1.3+1 [async aws_common built_collection built_value collection js meta path stack_trace stream_channel stream_transform]
- xdg_directories 0.2.0+3 [meta path process]
- xml 6.1.0 [collection meta petitparser]
- yaml 3.1.1 [collection source_span string_scanner]
```

### Device

iPhone 14 Pro Max Simulator

### OS

iOS 16.1

### Deployment Method

Amplify CLI

### CLI Version

10.6.2

### Additional Context

Tested on iOS only.

### Amplify Config

Instructions unclear

**Comment by HuiSF** [comment]
Thanks for opening this issue @vgribok I see an issue the underlying upload stream emitting `transferredBytes` as a whole but not uploaded chunk sizes, where the library adding the numbers up but not replacing it. Will open a PR to fix it soon.

**Comment by HuiSF** [comment]
The fix has been released with [`amplify_storage_s3_dart` 0.1.5](https://pub.dev/packages/amplify_storage_s3_dart/changelog), please run `flutter pub upgrade` to update packages to obtain the fix. Thanks.

**Comment by Jordan-Nelson** [comment]
Closing this issue out as it was resolved in the last release. Please open a new issue if you have any other troubles. Thanks.
